### PR TITLE
[MOS-357] Fix backing from new contact window

### PR DESCRIPTION
**Pull request by marziel**
Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/application-phonebook/windows/PhonebookNewContact.cpp [hunk 1]
```diff
@@ -132,11 +132,16 @@ namespace gui
...
             verifyAndSave();
-
-            return true;
+        }
+        else if (!inputEvent.isShortRelease(KeyCode::KEY_RF) || !shouldCurrentAppBeIgnoredOnSwitchBack()) {
+            return AppWindow::onInput(inputEvent);
         }
 
-        return ret;
+        return nameOfPreviousApplication.has_value()
```
Comment (rrandomsky) [inline module-apps/application-phonebook/windows/PhonebookNewContact.cpp:140]: Use `shouldCurrentAppBeIgnoredOnSwitchBack()` instead.
In _OptionContact.cpp : 51_ `ignoreCurrentWindowOnStack` flag is set to true, so is good to check it to.

This flag is in message is pass information "ignore a window" and `nameOfPreviousApplication` is just for help to switc back correctly - it's separate infos. Flag can be changed during "window flow" but name should stay the same.

TL;DR
In this case it's no matters, work fine fore sure, but for other window it can be important to check this flag too. I meet scenario when _Phonebook_ switch to _Messages_ and _Messages_ switch to _Templates_ and all of this windows have flag and nameOfPreviousApp but "window flow" looses this info....  and that's why `saveInfoAboutPreviousAppForProperSwitchBack();` in not on the beginning of `NewMessageWindow::onBeforeShow()` for example.
Using `shouldCurrentAppBeIgnoredOnSwitchBack()` is safe for future copy&paste of this mechanism to other Windows.

Comment (marziel) [reply module-apps/application-phonebook/windows/PhonebookNewContact.cpp]: Done

**Review by Lefucjusz — APPROVED** [review]

**Review by rrandomsky — APPROVED** [review]
